internal/utils/metrics/middleware.go

```diff
@@ -0,0 +1,76 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+)
+
+const (
+	MethodLabel = "method"
+)
+
+var (
+	RequestsInFlight = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "claudie_grpc_request_in_flight",
+			Help: "Number of grpc requests currently handled",
+		},
+	)
+
+	RequestCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "claudie_grpc_request_count",
+			Help: "Total number of gRPC API calls.",
+		},
+		[]string{MethodLabel},
+	)
+
+	Latency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "claudie_grpc_grpc_call_latency_seconds",
```

Comment:
Are there two `grpc` on purpose? I am currently working on the Grafana Dashboard and this just came up to my mind.